receiver/receiver.go

```diff
@@ -20,6 +20,29 @@ import (
 	_ "github.com/open-telemetry/opentelemetry-service/internal/compression/grpc" // load in supported grpc compression encodings
 )
 
+// Host represents the entity where the receiver is being hosted. It is used to
+// allow communication between the receiver and its host.
+type Host interface {
+	// Context returns a context provided by the host to be used on the receiver
+	// operations.
+	Context() context.Context
+
+	// AsyncErrorChannel returns a channel used by a receiver to notify its
+	// host about any fatal error (that requires re-starting the receiver) that
+	// that happen after its start function has returned.
+	AsyncErrorChannel() chan<- error
```

Comment:
AsyncErrorChannel() is never used for anything but sending values to it. Would it be nicer design to have a ReportError(err error) function instead? I think from receiver's perspective it is an implementation detailed that errors are reported via a channel inside the Host, so we may as well hide it and just expose a ReportError() func that is free to send it to a channel or do whatever else it wants.